File: apps/api/schemas/deliveries.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class DeliveryResponse(BaseModel):
    id: str
    source_kind: str
    source_run_id: str
    source_schedule_fire_id: str | None
    route_id: str
    status: str
    available_at: datetime
    attempt_count: int
    failure_code: str | None
    created_at: datetime
    updated_at: datetime
    delivered_at: datetime | None


class DeliveryPageResponse(BaseModel):
    items: list[DeliveryResponse]

File: packages/sdk-ts/src/resources/deliveries.ts
import type { FridayHttpClient } from "../http";

/** Safe status projection: delivery bodies and route credentials are absent. */
export interface Delivery {
  id: string;
  source_kind: string;
  source_run_id: string;
  source_schedule_fire_id: string | null;
  route_id: string;
  status: string;
  available_at: string;
  attempt_count: number;
  failure_code: string | null;
  created_at: string;
  updated_at: string;
  delivered_at: string | null;
}

export interface DeliveryPage {
  items: Delivery[];
}

export class DeliveriesResource {
  constructor(private readonly http: FridayHttpClient) {}

  listForRun(runId: string) {
    return this.http.request<DeliveryPage>({
      method: "GET",
      path: `/v1/runs/${runId}/deliveries`,
    });
  }
  get(deliveryId: string) {
    return this.http.request<Delivery>({
      method: "GET",
      path: `/v1/deliveries/${deliveryId}`,
    });
  }
  cancel(deliveryId: string) {
    return this.http.request<Delivery>({
      method: "POST",
      path: `/v1/deliveries/${deliveryId}/cancel`,
    });
  }
}
